File: backend/models/Dealer.js
const mongoose = require('mongoose');

const dealerSchema = new mongoose.Schema({
  localId: { type: Number, required: true, unique: true },
  dealerName: { type: String, required: true },
  place: { type: String, default: '' },
  phone: { type: String, default: '' },
  alternatePhone: { type: String, default: '' },
  email: { type: String, default: '' },
  address: { type: String, default: '' },
  notes: { type: String, default: '' },
  photoPath: { type: String, default: '' },
  createdAt: { type: Number, default: Date.now },
  updatedAt: { type: Number, default: Date.now }
});

module.exports = mongoose.model('Dealer', dealerSchema);

File: app/src/main/java/com/cocobiz/app/domain/model/Dealer.kt
package com.cocobiz.app.domain.model

data class Dealer(
    val id: Long = 0L,
    val dealerName: String,
    val place: String,
    val phone: String,
    val alternatePhone: String = "",
    val email: String = "",
    val address: String = "",
    val notes: String = "",
    val photoPath: String = "",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)

data class DealerWithStats(
    val dealer: Dealer,
    val totalRevenue: Double = 0.0,
    val activeSalesCount: Int = 0,
    val completedSalesCount: Int = 0,
    val lastSaleDate: Long? = null,
    val upcomingSaleDate: Long? = null
)
